PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp

```diff
@@ -165,24 +165,17 @@ LanguageNameInfo ScintillaEditView::_langNameInfoArray[L_EXTERNAL + 1] = {
 
 int getNbDigits(int aNum, int base)
 {
-	int nbChiffre = 1;
-	int diviseur = base;
+	int nbDigits = 0;
 
-	for (;;)
+	do
 	{
-		int result = aNum / diviseur;

```

Comment:
@cddiffz Did the division by zero happen here?

Reply:
Yes, the divisor can become zero due to an arithmetic overflow.

Reply:
ok, I see, because also in the new code a div by zero is possible if the provided base parameter is already zero by the caller. So I think an additional check for that makes sense.

Reply:
The actual calling code first sets the base to 10, then possibly changes it to either 2, 8 or 16. If zero were passed as the base, the calling code would be flawed anyway, wouldn't it? Trying to catch this makes no sense.